### DISCUSSION TITLE
creating a new Doc with POS values results in KeyError

### DISCUSSION THREAD

**Discussion by omri374** [explosion/spaCy#9145] [Help: Coding & Implementations] [answered]
I'm trying to create a `Doc` out of a pre-tokenized and pre-labeled sentence:

```python
sentence = [(('David', 'NNP'), 'O'),
 (('Pierce', 'NNP'), 'O'),
 (('52', 'CD'), 'O'),
 (('years', 'NNS'), 'O'),
 (('old', 'JJ'), 'O'),
 (('will', 'MD'), 'O'),
 (('join', 'VB'), 'O'),
 (('the', 'DT'), 'O'),
 (('board', 'NN'), 'O'),
 (('as', 'IN'), 'O'),
 (('a', 'DT'), 'O'),
 (('nonexecutive', 'JJ'), 'O'),
 (('director', 'NN'), 'O'),
 (('Dec.', 'NNP'), 'O'),
 (('12', 'CD'), 'O')]
```

The way I'm creating the `Doc` is the following:
```python
doc = Doc(vocab=nlp.vocab,
          words = [token[0][0] for token in sentence],
          pos = [token[0][1] for token in sentence],
          ents = [token[1] for token in sentence])
```

This seem to work, but whenever I try to extract the POS values out of the `Doc` I get a `KeyError` exception:

```python
[token.pos_ for token in doc]
```

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 1, in <listcomp>
  File "spacy\tokens\token.pyx", line 867, in spacy.tokens.token.Token.pos_.__get__
KeyError: -1059636719
```

Am I creating the `Doc` object in a wrong way? Can data similar to mine be transformed into a `Doc`? I've noticed that the addition of the `pos` field to the Doc object was introduced in version 3.

Thanks!

## Your Environment
- **spaCy version:** 3.1.2
- **Platform:** Windows-10-10.0.19043-SP0
- **Python version:** 3.9.6
- **Pipelines:** en_core_web_sm (3.1.0)

**Comment by polm** [comment, marked as answer]
spaCy tokens have two fields related to part of speech: `tag`, which contains anything, and `pos`, which is reserved for Universal Dependencies tags. The tags you are setting are not UD tags. When the doc is created they're happily set, but at lookup time they aren't found in the UD tag list and you get this error. The code for this is [here](https://github.com/explosion/spaCy/blob/632d8d4c356f424e0c9c18fc2edc1e75955586ad/spacy/tokens/token.pyx#L867), and the lookup table is [here](https://github.com/explosion/spaCy/blob/master/spacy/parts_of_speech.pyx). 

I actually personally hadn't realized the API was this restrictive, but it makes sense. However the tags should probably be validated when they're set and give an error that makes it clear what's going on, rather than blowing up when you access the doc like this.

In this case I think that if you just set your values as `tag` rather than `pos` you should be able to do what you want.

**Reply by omri374** [reply]
Thanks @polm! I agree that perhaps providing some guidance during the creation of the Doc could be more useful. Would you be interested in a PR for this?

**Reply by polm** [reply]
Thanks, I actually already started one: #9148. If you have any suggestions please feel free to chime in!
